code/game/objects/items/devices/radio/radio.dm

```diff
@@ -545,23 +545,22 @@
 	set_frequency(FREQ_SYNDICATE)
 
 /obj/item/radio/borg/screwdriver_act(mob/living/user, obj/item/tool)
-	if(!keyslot)
-		to_chat(user, span_warning("This radio doesn't have any encryption keys!"))
-		return
-
-	for(var/ch_name in channels)
-		SSradio.remove_object(src, GLOB.radiochannels[ch_name])
-		secure_radio_connections[ch_name] = null
+	if(keyslot && keyslot.canRemove)
+		for(var/ch_name in channels)
+			SSradio.remove_object(src, GLOB.radiochannels[ch_name])
+			secure_radio_connections[ch_name] = null
 
-	if(keyslot)
-		var/turf/user_turf = get_turf(user)
-		if(user_turf)
-			keyslot.forceMove(user_turf)
+		if(keyslot && keyslot.canRemove)
+			user.put_in_hands(keyslot)
 			keyslot = null
 
-	recalculateChannels()
-	to_chat(user, span_notice("You pop out the encryption key in the radio."))
-	return ..()
+		recalculateChannels()
+		to_chat(user, span_notice("You pop out the encryption key in the radio."))
+		return
+	else
+		to_chat(user, span_warning("This radio doesn't have any encryption keys!"))
```

Comment:
Same as above with "any removable keys"

Reply:
Alright, i added that here as well, shorten it to only keyslot since you cant insert anything into keyslot2 on the radio